test/extended/builds/ldap_fixtures.go

```diff
@@ -0,0 +1,90 @@
+package builds
+
+import (
+	"fmt"
+	"strings"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	buildapi "github.com/openshift/origin/pkg/build/api"
+	imageapi "github.com/openshift/origin/pkg/image/api"
+	exutil "github.com/openshift/origin/test/extended/util"
+)
+
+var _ = Describe("default: OpenLDAP build and deployment", func() {
+	defer GinkgoRecover()
+	var (
+		baseImageStreamName      = "openldap"
+		imageStreamFixture       = exutil.FixturePath("fixtures", "ldap", "ldapserver-imagestream.json")
+		imageStreamTargetFixture = exutil.FixturePath("fixtures", "ldap", "ldapserver-imagestream-testenv.json")
+		buildConfigFixture       = exutil.FixturePath("fixtures", "ldap", "ldapserver-buildconfig.json")
+		deploymentConfigFixture  = exutil.FixturePath("fixtures", "ldap", "ldapserver-deploymentconfig.json")
+		serviceConfigFixture     = exutil.FixturePath("fixtures", "ldap", "ldapserver-service.json")
+		oc                       = exutil.NewCLI("openldap", exutil.KubeConfigPath())
+	)
+
+	Describe("Building and deploying an OpenLDAP server", func() {
+		It(fmt.Sprintf("should create a image from %q template and run it in a pod", buildConfigFixture), func() {
+			oc.SetOutputDir(exutil.TestContext.OutputDir)
+
+			By(fmt.Sprintf("calling oc create -f %q", imageStreamFixture))
```

Comment:
> Why don't we want discrete steps here? That's more useful from a what-broke standpoint. no?

Marginally, but less likely to be used if we try to describe to someone how to run the example.  The only real benefit is you have corrupted json, but even so, your search area is small.